### PR TITLE
DATA-3184: Throw exception in StreamProcessor if data contract payload comes in after valid to date

### PR DESCRIPTION
**Pull request by m0h318**
This branch is branched out from the branch from #26
The changes to be reviewed starts from commit [added valid_from and valid_to checks](https://github.com/mhlabs/streamprocessor/commit/ee0f5a3198d5e1e76c528972f0794d2cbb4e5741)

### CODE REVIEW

File: core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java [hunk 1]
```diff
@@ -58,21 +71,48 @@ public void processElement(@Element PubsubMessage received, OutputReceiver<Pubsu
...
+                LocalDate validFromDate = LocalDate.parse(validFrom);
+
+                if (currentDate.isBefore(validFromDate)) {
+                    throw new CustomExceptionsUtils.InactiveDataContractException(
+                            "Data contract is not valid for the current time. "
+                                    + "Data contract is valid from: "
+                                    + validFrom);
+
+                } else if (dataContract.isNull("valid_to")) {
+                    String validTo = dataContract.getString("valid_to");
```
Comment (kramstrom) [inline core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java:90]: if `valid_to` is null then we shouldn't fetch it from the contract right?

Shouldn't it be something like:

```java
else if (!dataContract.isNull("valid_to")) {
  String validTo = dataContract.getString("valid_to");
  ...
}
```

Comment (m0h318) [reply core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java:90]: yes, thanks!

File: core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java [hunk 2]
```diff
@@ -73,6 +75,36 @@ public void processElement(@Element PubsubMessage received, MultiOutputReceiver
                             .getJSONObject("source")
                             .getString("provider");
 
+            LocalDate currentDate = LocalDate.now(ZoneId.of("Europe/Stockholm"));
```
[the remark this answers is not in the document]
Comment (m0h318) [reply core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java:78]: I was thinking of that, but when we add the date in the data contract are the producers referring to UTC of local time?

[the remark this answers is not in the document]
Comment (kramstrom) [reply core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java:78]: UTC :)

**Review by kramstrom — APPROVED** [review]

**Pull request merged**